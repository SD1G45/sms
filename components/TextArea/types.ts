import React from "react";

export interface TextAreaProps
  extends React.HTMLAttributes<HTMLTextAreaElement> {
  value: string;
  label?: string;
  error?: boolean;
  errorMessage?: string;
  className?: string;
  type?: string;
}
