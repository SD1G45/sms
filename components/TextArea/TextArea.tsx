import Link from "next/link";
import React from "react";
import {
  ErrorMessage,
  Label,
  LabelContainer,
  StyledLink,
  StyledTextArea,
} from "./styles";
import { TextAreaProps } from "./types";

const TextArea = React.forwardRef<HTMLTextAreaElement, TextAreaProps>(
  ({
    error,
    errorMessage,
    value,
    label,
    className,
    type,
    linkText,
    linkValue,
    height,
    ...other
  }) => {
    return (
      <>
        <LabelContainer>
          <Label>{label || ""}</Label>
          <Link href={linkValue || "/"}>
            <StyledLink>{linkText || ""}</StyledLink>
          </Link>
        </LabelContainer>
        <StyledTextArea
          type={type}
          error={error}
          height={height}
          {...other}
          value={value}
        />
        {error && errorMessage && <ErrorMessage>{errorMessage}</ErrorMessage>}
      </>
    );
  }
);

TextArea.displayName = "TextArea";
export default TextArea;
