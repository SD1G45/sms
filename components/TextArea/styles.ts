import styled from "styled-components";
import { TextAreaProps } from "./types";

export const LabelContainer = styled.div`
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

export const Label = styled.label`
  display: block;
  padding-bottom: 7px;
`;

export const StyledLink = styled(Label)`
  color: ${(props) => props.theme.colors.primary};
  cursor: pointer;
`;

export const StyledTextArea = styled.textarea<Omit<TextAreaProps, "label">>`
  padding: 12px;
  border: 1px solid
    ${(props) =>
      props.error ? props.theme.colors.error : props.theme.colors.lightGray};
  border-radius: 10px;
  width: 100%;
  margin-bottom: ${(props) => (props.error ? 8 : 0)}px;
`;

export const ErrorMessage = styled.p`
  color: ${(props) => props.theme.colors.error};
  font-size: 0.9rem;
  margin-left: 7px;
`;
