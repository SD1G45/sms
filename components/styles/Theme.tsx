import React from "react";
import { ThemeProvider } from "styled-components";
import ITheme from "./types";

const screenWidth = {
  minSmall: `(min-width: ${430}px)`,
  minMedium: `(min-width: ${730}px)`,
  minLarge: `(min-width: ${1024}px)`,
  maxSmall: `(max-width: ${430}px)`,
  maxMedium: `(max-width: ${730}px)`,
  maxLarge: `(max-width: ${1024}px)`,
};

const lightTheme: ITheme = {
  colors: {
    primary: "#4881F0",
    lightGray: "#BEC8D2",
    textPrimary: "#000000",
    error: "#FF7F7F",
  },
  screenWidth: { ...screenWidth },
};

interface ThemeProps {
  children: React.ReactNode;
}

const Theme: React.FC<ThemeProps> = ({ children }) => (
  <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>
);

export default Theme;
