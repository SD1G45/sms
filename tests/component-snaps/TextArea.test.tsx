import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import Theme from "../../components/styles/Theme";
import TextArea from "../../components/TextArea";

const steps = ["Step 1", "Step 2", "Step 3"];

describe("<TextArea/>", () => {
  it("renders correctly with label", () => {
    const tree = renderer.create(
      <Theme>
        <TextArea
          value="test"
        />
      </Theme>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with label", () => {
    const tree = renderer.create(
      <Theme>
        <TextArea
          label="Test"
          value="test"
        />
      </Theme>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with error", () => {
    const tree = renderer.create(
      <Theme>
        <TextArea
          label="Test"
          value="test"
          error={true}
          errorMessage="Test Error"
        />
      </Theme>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});