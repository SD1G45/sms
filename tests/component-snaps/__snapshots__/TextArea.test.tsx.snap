// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextArea/> renders correctly 1`] = `
.c0 {
  width: 100%;
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  display: block;
  padding-bottom: 7px;
}

.c3 {
  padding: 12px;
  border: 1px solid #BEC8D2;
  border-radius: 10px;
  width: 100%;
  margin-bottom: 0px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <label
      className="c2"
    >
      Test
    </label>
  </div>
  <textarea
    className="c3"
    id="test"
    rows={5}
    value="test"
  />
</div>
`;

exports[`<TextArea/> renders correctly with error 1`] = `
.c0 {
  width: 100%;
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  display: block;
  padding-bottom: 7px;
}

.c3 {
  padding: 12px;
  border: 1px solid #FF7F7F;
  border-radius: 10px;
  width: 100%;
  margin-bottom: 8px;
}

.c4 {
  color: #FF7F7F;
  font-size: 0.9rem;
  margin-left: 7px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <label
      className="c2"
    >
      Test
    </label>
  </div>
  <textarea
    className="c3"
    id="test"
    rows={5}
    value="test"
  />
  <p
    className="c4"
  >
    Test Error
  </p>
</div>
`;
